src/draw/lv_draw_vector.c

```diff
@@ -825,15 +812,29 @@ void lv_vector_clear_area(lv_vector_dsc_t * dsc, const lv_area_t * rect)
     lv_area_copy(&(new_task->dsc.scissor_area), rect);
 }
 
-void lv_draw_vector(lv_vector_dsc_t * dsc)
+void lv_draw_vector(lv_layer_t * layer, lv_vector_dsc_t * dsc, const lv_area_t * coords)
 {
     if(!dsc->tasks.task_list) {
         return;
     }
 
-    lv_layer_t * layer = dsc->layer;
+    lv_ll_t * task_list = dsc->tasks.task_list;
+    _lv_vector_draw_task * task = _lv_ll_get_head(task_list);
+    _lv_vector_draw_task * next_task = NULL;
+
+    while(task != NULL) {
+        next_task = _lv_ll_get_next(task_list, task);
+
+        if(coords) {
+            lv_matrix_translate(&task->dsc.matrix, coords->x1, coords->y1);
+        }
+
+        _lv_area_intersect(&task->dsc.scissor_area, &task->dsc.scissor_area, &layer->_clip_area);
+
+        task = next_task;
+    }
 
-    lv_draw_task_t * t = lv_draw_add_task(layer, &(layer->_clip_area));
+    lv_draw_task_t * t = lv_draw_add_task(layer, &layer->_clip_area);
```

Comment:
Here `coords` (the total area of the drawing) shall be used similarly to e.g. [`lv_draw_label()`](https://github.com/lvgl/lvgl/blob/master/src/draw/lv_draw_label.c#L85) 
```suggestion
    lv_draw_task_t * t = lv_draw_add_task(layer, coords);
```